gapis/api/transform2/transform.go

```diff
@@ -45,8 +49,15 @@ type Transform interface {
 	// Resources are needed for the state mutation, therefore this should be called after mutation.
 	ClearTransformResources(ctx context.Context)
 
-	// RequiresAccurateState returns true if transform needs the observe the accurate state.
+	// RequiresAccurateState returns true if the transform needs to observe the accurate state.
 	RequiresAccurateState() bool
+
+	// RequiresInnerStateMutation returns true if the transform needs to mutate state during the transformation
+	RequiresInnerStateMutation() bool
```

Comment:
I'm not sure about this name. It's pretty cryptic. What about ReadsStateBetweenCommands() or DoesYield()?

Reply:
It's not a read, it's actually a write operation on state. It would be a misleading name. For yield, I think it is already have a well known meaning and this really does not satisfy the meaning(as there is no actual yield).

We may change the name but I think it should be a significantly better name in terms of explaining what is happening.

Reply:
Again if we pass the state mutator in to TransformCommands() as an extra argument this can just go away. Then the name is irrelevant.

Reply:
Sorry, that comment is in reference to comments below!